fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -814,6 +814,8 @@ def forward(self, *args: Any, **kwargs: Any) -> torch.Tensor:
 
         if self.reshard_after_forward:
             self._free_full_params()
+            if self.mixed_precision:
+                self._free_fp16_param_shard()
```

Comment:
```suggestion
                # Save some memory by freeing fp16 shard since it will be recreated from
                # fp32 shard on backward. TODO (Min): may not need to free and recast
                # for the very last block.
                self._free_fp16_param_shard()
```

Reply:
never mind about the suggestion since it is merged. It is minor.